backend/slackbot/handler.js

```diff
@@ -1,91 +1,231 @@
+    console.log(`Error getting apps: ${error}`);
+    throw error;
   }
 
   // Get the statistics for those apps
   try {
     statistics = await getStatistics(apps);
   } catch (error) {
-    return {
-      statusCode: 500,
-      error: `Error getting stats: ${error}`
-    };
+    console.log(`Error getting stats: ${error}`);
+    throw error;
   }
 
+  // Send the stats to slack
   try {
-    await sendStats(statistics);
+    slackMessage = await sendStats(statistics);
   } catch (error) {
-    return {
-      statusCode: 500,
-      error: `Error sending to slack: ${error}`
-    };
+    console.log(`Error sending to slack: ${error}`);
+    throw error;
   }
 
-  return {
-    statusCode: 200, // OK
-    body: JSON.stringify({
-      response: statistics
-    })
-  };
+  return slackMessage;
 }
 
 /**
- *
- * @param gatewayURL the gateway url for aws lambda endpoint
- * @returns returns list of apps
+ * @param store the store (defaults to both unless specified)
+ * @returns returns list of apps (filtered by store if specified)
  */
-async function getApps() {
+async function getApps(store='both') {
   const endpoint = gatewayURL + '/apps';
-  return axios.get(endpoint);
+  let filteredApps = {};  // apps we want to return depending on store specified
+  store = store.toLowerCase();
+
+  const appsRequest = await axios.get(endpoint); // Get list of available apps
+  const apps = appsRequest.data;
+
+  // Filter apps based on if a store is specified
+  if (store == 'both') {
+    return apps;
+  } else {
+
+    for (let key in apps) {
+
+      if (store.includes('google') || store.includes('android')) {
+        if (key.includes('Google Play')) {
+          filteredApps[key] = apps[key];
+        }
+      }
+      else if (store.includes('apple') || store.includes('ios') || store.includes('app')) {
+        if (key.includes('App Store')) {
+          filteredApps[key] = apps[key];
+        }
+      }
+    }
 
+  }
+
+  return filteredApps;
 }
 
 /**
- *
- * @param applications the list of apps we want statistics from
+ * @param apps the list of apps we want statistics from
+ * @param parameters any optional or required parameters passed
  * @returns statsList a list of statistics for each specified version of an app
  */
-async function getStatistics(applications){
+async function getStatistics(apps, parameters={}){
   let promises = [];  // Array for promises
   let statsList = [];
 
-  const apps = applications.data;
   const endpoint = gatewayURL + '/stats';
 
   for (let app in apps){
     let appData = apps[app];
 
     // Using last 7 days
+    let days = 7;
+    if (parameters.hasOwnProperty('days')) {
+      days = parameters.days;
+    }
+
+    // Calculate date range using days or defaulting to 7 days back from latest available
     let endDate = new Date();
     let startDate = new Date();
-    startDate.setDate(endDate.getDate() - 7);
 
-    // Use the latest version currently
+    startDate.setDate(endDate.getDate() - days);
```

Comment:
Okay - that's definitely something we should clarify with Ford. It's totally okay if we need to handle both cases, but I'd prefer to do it more cleanly than we're doing here, either using separate functions or comments to explain further. 

A better way would be to detect whether we should look for `days` or `startDate`/`endDate` based on the command being processed (rather than ignoring the context and trying to find them both for all commands).